metricbeat/module/kafka/partition/partition.go

```diff
@@ -59,64 +54,42 @@ var debugf = logp.MakeDebug("kafka")
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	cfgwarn.Beta("The kafka partition metricset is beta")
 
-	config := defaultConfig
-	if err := base.Module().UnpackConfig(&config); err != nil {
-		return nil, err
-	}
-
-	var tls *tls.Config
-	tlsCfg, err := tlscommon.LoadTLSConfig(config.TLS)
+	ms, err := kafka.NewMetricSet(base)
 	if err != nil {
 		return nil, err
 	}
-	if tlsCfg != nil {
-		tls = tlsCfg.BuildModuleConfig("")
-	}
 
-	timeout := base.Module().Config().Timeout
-	cfg := kafka.BrokerSettings{
-		MatchID:     true,
-		DialTimeout: timeout,
-		ReadTimeout: timeout,
-		ClientID:    config.ClientID,
-		Retries:     config.Retries,
-		Backoff:     config.Backoff,
-		TLS:         tls,
-		Username:    config.Username,
-		Password:    config.Password,
-		Version:     kafka.Version("0.8.2.0"),
+	config := struct {
+		Topics []string `config:"topics"`
```

Comment:
Is this a shared config with consumergroup?

Reply:
Yes, I guess that if someone wants to monitor different topics in different metricsets they need to use different module definitions.

Reply:
do you expect all metricsets to have this config. if yes I think we should move it to the kafka metricset definition, if not each should prefix it's own topics config.

Yes, same story as above, would be a breaking change.

Reply:
Current metricsets have all this config, but for example a `broker` metricset wouldn't have it. I am thinking about adding it, because with current metricsets, a `broker` without replicas or consumers wouldn't appear at all in the events.